### PR TITLE
feat: improving pom.xml configuration, adding source codes to built dependency + removing directory profile

### LINKED ISSUE #95: The built mvn dependency should include the source code as well (make the debug and troubleshoot easier)
Right now if I build the project with:
`mvn clean install`

And then I add this dependency to a project and navigate to the form-filler-addon classes, then I will not have the source code:
 image

Investigate and make sure that the source code is part of the dependency as well!

### PR DESCRIPTION
**Pull request by czp13**
## Description
Improvements in the pom.xml:
- Removed directory profile (as it would be needed to upload code to the addon directory?)
- Adding build config to pom.xml to attach source code to the built dependency.

[Added maven-javadoc-plugin to <build></build> root, to have javadocs.](https://github.com/vaadin/form-filler-addon/pull/96/commits/a0a17be1523b7e564683e29857a1fe6a9694de59) 
[a0a17be](https://github.com/vaadin/form-filler-addon/pull/96/commits/a0a17be1523b7e564683e29857a1fe6a9694de59)

Other plugins:
- source plugin was added already,
- assembly we discussed (on the daily) to not add,
- maven-jar was added already with a slightly different configuration,

Thanks for the hints/ideas here: @mshabarov  and @mgarciavaadin 

Fixes #95 
- https://github.com/vaadin/form-filler-addon/issues/95

## Type of change

- [x] Bugfix
- [x] Feature

## Checklist

- [x] I have read the contribution guide: https://vaadin.com/docs/latest/guide/contributing/overview/
- [x] I have added a description following the guideline.
- [x] The issue is created in the corresponding repository and I have referenced it.
- [ ] I have added tests to ensure my change is effective and works as intended.
- [ ] New and existing tests are passing locally with my change.
- [x] I have performed self-review and correct misspellings.

#### Additional for `Feature` type of change

- [ ] Enhancement / new feature was discussed in a corresponding GitHub issue and Acceptance Criteria were created.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -207,76 +221,6 @@
...
-                            <execution>
-                                <id>attach-sources</id>
-                                <phase>verify</phase>
-                                <goals>
-                                    <goal>jar-no-fork</goal>
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
-                    <plugin>
```
Comment (czp13) [inline pom.xml:247]: A question here that should we include the:
- javadocs part(maven-javadoc-plugin)
- maven-jar-plugins part
to the build config? (<build>)?

File: pom.xml [hunk 2]
```diff
@@ -207,76 +221,6 @@
...
-                        <groupId>org.apache.maven.plugins</groupId>
-                        <artifactId>maven-javadoc-plugin</artifactId>
```
Comment (mshabarov) [inline pom.xml:249]: Javadoc and jar plugins might be needed, why you want to delete them?

Comment (mshabarov) [reply pom.xml:249]: I think they are both needed. But not in the directory profile - they have to run always.

Comment (mshabarov) [reply pom.xml:249]: Also directory profile seems not needed at all - we are not going to deploy this plugin into Vaadin Directory.

Comment (czp13) [reply pom.xml:249]: Thanks for the review! I may miss something :), but I see that I removed the `directory` profile fully from the `pom.xml. `

And I am on the same side about the removed plugins from the `directory` profile, as my comment shows above: 

I am thinking also about adding `javadoc` and `jar` plugins profiles back, and those could be added under the <build> without any <profile> for always running.

Let's catchup on this tomorrow daily. Take care until then!

Comment (mshabarov) [reply pom.xml:249]: Ah, indeed, you removed it. Sorry, it was the end of the day :)

File: pom.xml [hunk 3]
```diff
@@ -207,76 +221,6 @@
...
-                    </plugin>
-                    <plugin>
-                        <groupId>org.apache.maven.plugins</groupId>
-                        <artifactId>maven-jar-plugin</artifactId>
-                        <version>${maven.jar.plugin.version}</version>
-                        <configuration>
-                            <!-- Generated file that shouldn't be included in add-ons -->
-                            <excludes>
-                                <exclude>META-INF/VAADIN/config/flow-build-info.json</exclude>
-                                <exclude>META-INF/VAADIN/webapp/**/*</exclude>
```
Comment (czp13) [inline pom.xml:273]: Ok, we have this part in the build, check line 120/134.
 and line 153 does not contain the `  <exclude>META-INF/VAADIN/webapp/**/*</exclude>` 
 So if no objection then I will not add this line, and have the original `maven-jar-plugin configuration` in place.

Comment (mshabarov) [reply pom.xml:273]: Currently, form filler add-on has nothing to be placed into `VAADIN/webapp`.
Usually it's a bundle static resources or other static resource provided by Vaadin, that are placed there (e.g. JavaScript for Push).
However, if we place this exclusion and in the future we add some resources into add-on, we might struggle why it doesn't work properly.
So I think we can skip adding it at the moment.

**Review by mshabarov — APPROVED** [review]

File: pom.xml (hunk 3 above)
Comment (czp13) [reply pom.xml:273]: Thank you @mshabarov for your insights here and in the daily as well. It makes sense!
Also thanks for approve and merge :)

**Pull request merged**